content/en/docs/v3.5/learning/persistent-storage-files.md

```diff
@@ -0,0 +1,619 @@
+    </td>
+  </tr>
+
+  <tr>
+    <td><pre>./member/wal/0.tmp (or .../1.tmp)</pre></td>
+    <td>
+      <strong>Preallocated space</strong> for the next write ahead log file.
+      Used to avoid Raft being stuck by a lack of WAL logs capacity without the possibility to raise an alarm.
+    </td>
+  </tr>
+</table>
+
+
+### Temporary files
+
+During etcd internal processing, it is possible that several short living files might be encountered:
+
+<table>
+  <tr>
+    <th>File</th>
+    <th>High level purpose</th>
+  </tr>
+  <tr>
+    <td><pre>./member/snap/0000000000000002-000000000007a178.snap.broken</pre></td>
+    <td>
+      <p>Snapshot files are renamed as ‘broken’ when they cannot be <a href="https://github.com/etcd-io/etcd/blob/aa97484166d2b3fb6afeb4390344e68b02afb566/server/etcdserver/api/snap/snapshotter.go#L144">loaded</a>.<p>
+      <!-- TODO: etcdserver/api/snap/snapshotter.go:148 -->
+      <p>The attempt to load the newest file happens when etcd is being <a href="https://github.com/etcd-io/etcd/blob/aa97484166d2b3fb6afeb4390344e68b02afb566/server/etcdserver/server.go#L302">started</a>.</p>
+      <!-- TODO: etcdserver/server.go:428 -->
+      <p>Or during backup/migrate commands of etcdctl.</p>
+   </td>
+  </tr>
+  <tr>
+    <td><pre>./member/snap/tmp071677638 (random suffix)</pre></td>
+    <td>
+      <p><a href="https://github.com/etcd-io/etcd/blob/ae9734ed278b7a1a7dfc82e800471ebbf9fce56f/etcdserver/api/snap/db.go#L39">Temporary</a> (bbolt) file created on replicas in response to the msgSnap leaders request, so to the demand from the leader to recover storage from the given snapshot.</p>
+      <p>
+        After successful (complete) retrieval of content the file is <a href="https://github.com/etcd-io/etcd/blob/ae9734ed278b7a1a7dfc82e800471ebbf9fce56f/etcdserver/api/snap/db.go#L55">renamed</a> to: <code>/member/snap/[SNAPSHOT-INDEX].snap.db</code>. In case of a server dying / being killed in the middle of the files download, the files remain on disk and are never automatically cleaned.They can be substantial in size (GBs).
+      </p>
+      <p>See <a href="https://github.com/etcd-io/etcd/issues/12837">etcd/issues/12837</a>. <strong>Fixed in etcd 3.5.</strong></p>
+    </td>
+  </tr>
+  <tr>
+    <td><pre>/member/snap/db.tmp.071677638 (random suffix)</pre></td>
+    <td>
+      <p>A temporary file that contains a copy of the backend content (/member/snap/db), during the <a href="https://github.com/etcd-io/etcd/blob/a4570a60e771402360755beb7d662bdbca1f87f2/server/mvcc/backend/backend.go#L373">process of defragmentation</a>. After the successful process the file is renamed to /member/snap/db, replacing the original backend.</p>
+      <p>On etcd server startup these files get<a href="https://github.com/etcd-io/etcd/blob/a4570a60e771402360755beb7d662bdbca1f87f2/server/etcdserver/api/snap/snapshotter.go#L217"> pruned</a>.</p>
+    </td>
+  </tr>
+</table>
+
+## Bbolt b+tree: **member/snap/db**
```

Comment:
I think using `bbolt` instead of `Bbolt` would be good. That's how we do in the bbolt repo as well.